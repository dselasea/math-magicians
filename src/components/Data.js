import { useEffect, useState } from 'react';

const Data = () => {
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(false);
  const [quotes, setQuotes] = useState([]);

  useEffect(() => {
    const fetchQuotes = async () => {
      setLoading(true);
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=intelligence', {
          method: 'GET',
          headers: {
            'X-Api-Key': 'EhL2F3KlXGZe2HF8Zy+j8Q==UAJsJ1pJRqnCBQb9',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(),
        });
        const data = await response.json();
        setQuotes(data);
      } catch (error) {
        setError(true);
      }
      setLoading(false);
    };
    fetchQuotes();
  }, []);
  if (error) return <div>Something went wrong</div>;
  if (loading) return <div>Loading...</div>;
  return (
    <div>
      {
        quotes.map((quote) => (
          <div key={quote.category}>
            <h1>{quote.author}</h1>
            <p>
              {quote.quote}
              {' '}
              <span>
                ~
                {quote.category}
              </span>
            </p>
          </div>
        ))
      }
    </div>
  );
};

export default Data;
