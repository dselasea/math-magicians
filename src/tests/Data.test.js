import renderer from 'react-test-renderer';
import Data from '../components/Data';

test('Test Data Component', () => {
  const component = renderer.create(<Data />);
  const data = component.toJSON();
  expect(data).toMatchSnapshot();
});
