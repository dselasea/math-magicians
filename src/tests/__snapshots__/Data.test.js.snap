// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Data Component 1`] = `null`;
